eox_nelp/api_clients/tests/__init__.py

```diff
@@ -177,3 +141,80 @@ def test_failed_get(self, auth_mock, requests_mock):
         self.assertEqual(logs.output, [
             f"ERROR:{api_clients.__name__}:{log_error}"
         ])
+
+
+class TestOauth2ApiClientMixin(TestApiClientMixin):
+    """
+    This test class contains test cases for the `AbstractOauth2ApiClient` class
+    to ensure that the authentication process using OAuth2 is working correctly.
+    """
+
+    def test_failed_authentication(self):
+        """Test case for invalid credentials.
+
+        Expected behavior:
+            - Raise MissingTokenError exception
+        """
+        self.assertRaises(MissingTokenError, self.api_class)
+
+    @patch("eox_nelp.api_clients.OAuth2Session")
+    def test_successful_authentication(self, oauth2_session_mock):
+        """Test case when the authentication response is valid.
+
+        Expected behavior:
+            - Session is set
+            - Session headers contains Authorization key.
+            - fetch_token was called with the right values.
+        """
+        fetch_token_mock = Mock()
+        fetch_token_mock.return_value = {
+            "token_type": "Bearer",
```

Comment:
I was checking and this auth only work with bearer. In the future could be considered the jwt way or possibility.

Reply:
I found [here](https://github.com/eduNEXT/eox-nelp/blob/3efc7316a8a0b66a08ec5b73264c0d0ba0070401/eox_nelp/api_clients/__init__.py#L148), that the kind of token depends on the response

Reply:
In edx-platform you can ask the type. By the way, is ok.
![image](https://github.com/eduNEXT/eox-nelp/assets/51926076/2c46089e-4cd0-4b19-bbb2-3bb4b20479b8)

Reply:
keep in mind that this was initially designed  to consume the futurex service and probably won't fit in different different situations